datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java

```diff
@@ -53,7 +58,12 @@ public void createUser(@RequestBody UserDTO requestDTO) {
     @ApiOperation("Delete user")
     public void deleteUser(@RequestHeader String user,
                            @RequestHeader String password) {
-        UserIDAuth userIDAuth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
-        dataSafeService.userProfile().deregister(userIDAuth);
+        UserIDAuth auth = new UserIDAuth(new UserID(user), new ReadKeyPassword(password));
+
+        if (dataSafeService.userProfile().userExists(auth.getUserID()))
+            dataSafeService.userProfile().deregister(auth);
+        else
```

Comment:
Makes sense to invert check (to reduce nestedness), also please use if with curly braces (see https://stackoverflow.com/questions/2125066/is-it-a-bad-practice-to-use-an-if-statement-without-curly-braces):
```
if (!dataSafeService.userProfile().userExists(auth.getUserID())) {
    throw new UserDoesNotExists("User \"" + auth.getUserID().getValue() + "\" does not exists");
}

dataSafeService.userProfile().deregister(auth);
```

Reply:
ok.